core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java

```diff
@@ -51,14 +65,20 @@ public AnnexAwareDeclarativeLabelProvider(ILabelProvider delegate) {
 	protected Object doGetText(Object element) {
 		ILabelProvider labelProvider = getLabelProvider(element);
 		PolymorphicDispatcher<Object> textDispatcher = new PolymorphicDispatcher<Object>("text", 1, 1,
-				Collections.singletonList(labelProvider), new ErrorHandler<Object>() {
-					@Override
-					public Object handle(Object[] params, Throwable e) {
-						return handleTextError(params, e);
-					}
-				});
+				Collections.singletonList(labelProvider), (params, e) -> handleTextError(params, e));
 		Object text = textDispatcher.invoke(element);
 		if (text != null) {
+			if (element instanceof SystemImplementation || element instanceof DataImplementation
+					|| element instanceof ProcessorImplementation || element instanceof ProcessImplementation
+					|| element instanceof ThreadGroupImplementation || element instanceof ThreadImplementation
+					|| element instanceof SubprogramImplementation || element instanceof SubprogramGroupImplementation
+					|| element instanceof AbstractImplementation || element instanceof BusImplementation
+					|| element instanceof MemoryImplementation || element instanceof DeviceImplementation
+					|| element instanceof VirtualBusImplementation || element instanceof VirtualProcessorImplementation
+					|| element instanceof FlowImplementation) {
+
```

Comment:
As you list ALL sublasses of ComponentImplementation here, you can replace the condition with if (element instanceof ComponentImplementation)

Reply:
changed to ComponentImplementation